app/Controllers/CopyController.php

```
@@ -290,6 +290,73 @@ public function updateCopy(Request $request, Response $response, mysqli $db, int
+            $repo->create($bookId, $numero, $stato, $note !== '' ? $note : null);
+        } catch (\Throwable $e) {
+            SecureLogger::error('[CopyController] createCopy failed', ['book' => $bookId, 'error' => $e->getMessage()]);
+            $_SESSION['error_message'] = __('Impossibile aggiungere la copia.');
+            return $response->withHeader('Location', url("/admin/books/{$bookId}"))->withStatus(302);
+        }
+
+        (new \App\Support\DataIntegrity($db))->recalculateBookAvailability($bookId);
+
+        $_SESSION['success_message'] = __('Copia aggiunta con successo.');
+        return $response->withHeader('Location', url("/admin/books/{$bookId}"))->withStatus(302);
+    }
```

PHPMD (2.15.0)

[error] createCopy accesses the super-global variable $_SESSION. (undefined)

(Superglobals)